src/Helper/HeaderHelper.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Helper;
+
+use Psr\Http\Message\RequestInterface;
+
+final class HeaderHelper
+{
+    /**
+     * Explode header value to value and parameters (eg. text/html;q=2;version=6)
+     * @param string $headerValue
+     * @return array first element is the value, and key-value are the parameters
+     */
+    public static function getValueAndParameters(string $headerValue): array
+    {
+        $headerValue = trim($headerValue);
+        if ($headerValue === '') {
+            return [];
+        }
+        $parts = preg_split('/\s*;\s*/', $headerValue, -1, PREG_SPLIT_NO_EMPTY);
+        $output = [array_shift($parts)];
+        foreach ($parts as $part) {
+            [$key, $headerValue] = explode('=', $part, 2);
+            $output[$key] = $headerValue;
+        }
+        return $output;
+    }
+
+    /**
+     * Getting header value as q factor sorted list
+     * @param string|string[] $values Header value as a comma-separated string or already exploded string array.
+     * @see getValueAndParameters
+     * @link https://developer.mozilla.org/en-US/docs/Glossary/Quality_values
+     */
+    public static function getByQFactorSortedList($values): array
```

Comment:
getValuesListSortedByQFactor?

Reply:
`getSortedValueAndParameters`

No other sorting method found. What I have found is all sorted by q factor, so it's unnecessary to name q factor in the method name.